packages/ethereum-contracts/contracts/superfluid/SuperToken.sol

```diff
@@ -192,7 +193,7 @@ contract SuperToken is
      * For more context, see https://github.com/superfluid-finance/protocol-monorepo/wiki/About-ERC-777
      */
     function _transferFrom(address spender, address holder, address recipient, uint amount)
-        internal returns (bool)
+        internal virtual returns (bool)
```

Comment:
We are doing a much bigger surgery than I heard of.

This is code smell and anti-pattern, in general one should not change the behaviour of "_" for it is crafted logic for the reusing site.

Instead, a more idiomatic OO way (not really promoting it anymore, but within the conext of this codebase) is to override the site that uses those code, rewiring the logic etc.

Reply:
let's discuss it separately tomorrow.

Reply:
changed to only have external and public methods virtual